app/src/main/java/cm/aptoide/pt/editorialList/EditorialListPresenter.java

```diff
@@ -211,4 +203,94 @@ public EditorialListPresenter(EditorialListView editorialListView,
+        .subscribe(lifecycleEvent -> {
+        }, crashReporter::log);
+  }
+
+  @VisibleForTesting public void handleUserReaction() {
+    view.getLifecycleEvent()
+        .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.reactionClicked())
+        .flatMap(
+            reactionsHomeEvent -> editorialListManager.setReaction(reactionsHomeEvent.getCardId(),
+                reactionsHomeEvent.getGroupId(), reactionsHomeEvent.getReaction())
+                .toObservable()
+                .filter(ReactionsResponse::differentReaction)
+                .observeOn(viewScheduler)
+                .doOnNext(reactionsResponse -> handleReactionsResponse(reactionsResponse, false))
+                .filter(ReactionsResponse::wasSuccess)
+                .flatMapSingle(__ -> loadReactionModel(reactionsHomeEvent.getCardId(),
+                    reactionsHomeEvent.getGroupId())))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(lifecycleEvent -> {
+        }, crashReporter::log);
+  }
+
+  private void handleReactionsResponse(ReactionsResponse reactionsResponse, boolean isDelete) {
+    if (reactionsResponse.wasSuccess()) {
+      if (isDelete) {
+        editorialListAnalytics.sendDeleteEvent();
+      } else {
+        editorialListAnalytics.sendReactedEvent();
+      }
+    } else if (reactionsResponse.reactionsExceeded()) {
+      view.showLogInDialog();
+    } else if (reactionsResponse.wasNetworkError()) {
+      view.showNetworkErrorToast();
+    } else if (reactionsResponse.wasGeneralError()) {
+      view.showGenericErrorToast();
+    }
+  }
+
+  @VisibleForTesting public void handleSnackLogInClick() {
+    view.getLifecycleEvent()
+        .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.snackLogInClick())
+        .doOnNext(homeEvent -> editorialListNavigator.navigateToLogIn())
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(lifecycleEvent -> {
+        }, crashReporter::log);
+  }
+
+  private Observable<CurationCard> singlePressReactionButtonAction(
+      EditorialHomeEvent editorialHomeEvent) {
+    boolean isFirstReaction = editorialListManager.isFirstReaction(editorialHomeEvent.getCardId(),
+        editorialHomeEvent.getGroupId());
+    if (isFirstReaction) {
+      showReactions(editorialHomeEvent);
+      return Observable.just(new CurationCard());
+    } else {
+      return editorialListManager.deleteReaction(editorialHomeEvent.getCardId(),
+          editorialHomeEvent.getGroupId())
+          .toObservable()
+          .doOnNext(reactionsResponse -> handleReactionsResponse(reactionsResponse, true))
+          .filter(ReactionsResponse::wasSuccess)
+          .flatMapSingle(reactionsResponse -> loadReactionModel(editorialHomeEvent.getCardId(),
+              editorialHomeEvent.getGroupId()));
```

Comment:
This would have the same comment as the HomePresenter singlePressReactionButtonAction method